Upgrade to latest navigation library version, use FragmentContainerView

AmandaRiu opened the pull request:
Possibly fixes #1719 by using the FragmentContainerView instead of the fragment element as the navigation host container to ensure fragment transactions are properly handled and no timing issues
creep up. The `FragmentContainerView` was broken in `2.2.0-rc02` version of the androidx navigation library ([see bug](https://issuetracker.google.com/issues/143752103)), but was fixed in `2.2.0-rc03`. This PR does the following:
- upgrades the androidx.navigation library to `2.2.0-rc03`
- swaps out the use of `<fragment/>` as the navhost container with `<FragmentContainerView/>`

Since I was never able to reproduce the issue I'm unsure how to test if this will _fix_ things, but at the very least it's the proper implementation of the feature so now Google fixed we should use it. 😏 

### Recommended Testing Areas
- Test opening child fragments and performing configuration changes (example - Orders -> Order Detail -> Order Fulfillment -> minimize app and reopen) : "Don't keep activities" should be enabled in dev options. 
- Test clicking through setting screens and rotating the device.

Commit:
Upgrade to latest navigation library version, use FragmentContainerView

Use the FragmentContainerView instead of the fragment element to ensure
the fragment transactions are properly handled and no timing issues
creep up. Possibly fixes #1719

nbradbury commented on build.gradle:
```diff
@@ -1,6 +1,6 @@
 buildscript {
     ext.kotlinVersion = '1.3.60'
-    ext.navigationVersion = '2.2.0-rc02'
+    ext.navigationVersion = '2.2.0-rc03'
```
The latest release is [2.2.0-rc04](https://developer.android.com/jetpack/androidx/releases/navigation). Should we be using that instead?

nbradbury replied:
Related to this, should we also update to the latest [AndroidX fragment library](https://developer.android.com/jetpack/androidx/releases/fragment)?

AmandaRiu replied:
Great catch! Fixed in 1dfd50e9e450ddf5321e3fe108d07ca9742b07d3

nbradbury commented:
@anitaa1990 I'll go ahead and take over this PR since you're AFK.

nbradbury commented:
When I run this build, I get the following exception:

```
java.lang.IllegalStateException: Activity com.woocommerce.android.ui.main.MainActivity@cf9a703 does not have a NavController set on 2131362374
        at androidx.navigation.Navigation.findNavController(Navigation.java:61)
        at androidx.navigation.ActivityKt.findNavController(Activity.kt:30)
        at com.woocommerce.android.ui.main.MainActivity.onCreate(MainActivity.kt:129)
```

This is happening on multiple emulators, and upgrading to 2.2.0-rc04 made no difference.

Commit:
Update to latest release candidate version

Commit:
Move the navController initialization to onPostCreate()

This is due to a known issue with FragmentContainerView as documented
in this Google bug tracker ticket:
https://issuetracker.google.com/issues/142847973

The work around is to get the navController in onPostCreate()

AmandaRiu commented:
@nbradbury very bizarre about that crash. It was working when I tested the PR, but I did start with changing the host fragment in `activity_app_settings` first, and since that's not the default startup, it works without any further changes. It's the `activity_main` change that's causing the crash and this is because of the timing during app launch. I found [this Google bugtracker issue](https://issuetracker.google.com/issues/142847973) that explains why the crash happens and it's marked as "wontfix". That ticket does list a couple options for a workaround and the one to move the fetching of the `navController` from `onCreate` to `onPostCreate` is what fixed the crash for me. That fix can be found here a729fef3594a279c64c08694b5d96d976336eb2f

Ready for another round.

nbradbury reviewed (approved):
:shipit:
